modules/apps/foundation/users-admin/users-admin-web/src/main/java/com/liferay/users/admin/web/util/comparator/OrganizationMemberNameComparator.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.users.admin.web.util.comparator;
+
+import com.liferay.portal.kernel.model.Organization;
+import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.util.OrderByComparator;
+
+/**
+ * @author Pei-Jung Lan
+ */
+public class OrganizationMemberNameComparator
+	extends OrderByComparator<Object> {
+
+	public static final String ORDER_BY_ASC = "name ASC";
+
+	public static final String ORDER_BY_DESC = "name DESC";
+
+	public static final String[] ORDER_BY_FIELDS = {"name"};
+
+	public OrganizationMemberNameComparator() {
+		this(false);
+	}
+
+	public OrganizationMemberNameComparator(boolean ascending) {
+		_ascending = ascending;
+	}
+
+	@Override
+	public int compare(Object object1, Object object2) {
+		String name1;
+		String name2;
+
+		if (object1 instanceof Organization) {
```

Comment:
The portal-kernel model package contains a lot of `@ProviderType` models that are provided by other modules. Modules that extend interfaces in portal-kernel model are **providers** and have a narrower import range of [minor, minor+1). So even a minor version change to the package requires every provider type bundles that uses the package to import the new package, and we need to manually updates these bundles. Otherwise, they cannot be resolved due to potential incompatibility with the available package. 

This is an issue we've encountered several times. So far, we've tried to avoid making changes to portal-kernel model, and have the changes in other places if possible. I understand that this is not the best approach, but it's the best we can do now. Speaking of which, do we have any plan to modularize this package and migrate its classes to new modules in the near future?